src/javascripts/components/tabs.js

```diff
@@ -1,133 +1,132 @@
+        $tabTrigger.parentNode.classList.remove(tabsItemCurrentClass)
+      } else if ($tabTrigger.parentNode.classList.contains(headingItemCurrentClass)) {
+        $tabTrigger.parentNode.classList.remove(headingItemCurrentClass)
       }
-    } else {
-      // Reset tabs
-      Tabs.resetTabs(id)
-
-      // Set current active tab for both tabs and accordion links
-      $target.attr('aria-expanded', 'true')
-      $.each($targetParent, function (key, obj) {
-        if ($(obj).hasClass('app-tabs__item')) {
-          $(obj).addClass('app-tabs__item--current')
-        } else if ($(obj).hasClass('app-tabs__heading')) {
-          $(obj).addClass('app-tabs__heading--current')
-        }
-      })
-
-      // Set current container
-      $(id).show().attr('aria-hidden', 'false')
-    }
-  },
-
-  // Tabs mode
-  clickTabItem: function (e) {
-    e.preventDefault()
-
-    // Get tab container ID
-    var id = $(this).attr('href')
-
-    // Activate current tab
-    Tabs.activateAndToggleCurrentTab(id)
-  },
-
-  // Close current container on click
-  clickCloseContainer: function (e) {
-    e.preventDefault()
-
-    // Get tab container ID and save example object
-    var id = $(this).parent().attr('id')
-
+    })
+  } else {
     // Reset tabs
-    Tabs.resetTabs('#' + id)
-  },
-
-  init: function (selector) {
-    // If more than one container
-    var $examples = $(selector)
-    $.each($examples, function (key, obj) {
-      var tabsContainer = $(obj).find('.js-tabs__container')
-      if (tabsContainer.length > 0 && !tabsContainer.hasClass('js-tabs__container--no-tabs')) {
-        // Hide all containers
-        tabsContainer.hide()
-        Tabs.attachAriaAttributes(tabsContainer)
-        // Add close button to each container
-        tabsContainer.append('<button class="app-tabs__close js-tabs__close">Close</button>')
-        tabsContainer.addClass('app-tabs__container--with-close-button')
+    this.resetTabs()
+    // make current active
+    $tabContent.classList.remove(tabContainerHiddenClass)
+    $tabContent.setAttribute('aria-hidden', 'false')
+
+    nodeListForEach($currentTargetSiblings, function ($tabTrigger) {
+      $tabTrigger.setAttribute('aria-expanded', 'true')
+      if ($tabTrigger.parentNode.classList.contains(tabsItemClass)) {
+        $tabTrigger.parentNode.classList.add(tabsItemCurrentClass)
+      } else if ($tabTrigger.parentNode.classList.contains(headingItemClass)) {
+        $tabTrigger.parentNode.classList.add(headingItemCurrentClass)
       }
     })
+  }
+}
+// reset aria attributes to default and close the tab content container
+AppTabs.prototype.resetTabs = function () {
+  nodeListForEach(this.$allTabsContainers, function ($tabContainer) {
+    if (!$tabContainer.classList.contains(tabContainerNoTabsJsClass)) {
+      $tabContainer.classList.add(tabContainerHiddenClass)
+      $tabContainer.setAttribute('aria-hidden', 'true')
+    }
+  })
+
+  nodeListForEach(this.$tabTriggers, function ($tabTrigger) {
```

Comment:
Non-blocking: There is a lot of looping that when you look at the GOV.UK Frontend tabs they don't seem to be able to avoid

Reply:
i could look at tab containers length and use that loop to modify triggers, but i'm not sure that's a better way to do it